components/brave_wallet/browser/asset_ratio_service.cc

```diff
@@ -154,6 +170,53 @@ GURL AssetRatioService::GetPriceHistoryURL(
   return GURL(spec);
 }
 
+void AssetRatioService::GetBuyUrlV1(mojom::OnRampProvider provider,
+                                    const std::string& chain_id,
+                                    const std::string& address,
+                                    const std::string& symbol,
+                                    const std::string& amount,
+                                    const std::string& currency_code,
+                                    GetBuyUrlV1Callback callback) {
+  std::string url;
+  if (provider == mojom::OnRampProvider::kWyre) {
+    url =
+        base::StringPrintf(kWyreBuyUrl, address.c_str(), currency_code.c_str(),
+                           symbol.c_str(), amount.c_str(), kWyreID);
+    std::move(callback).Run(std::move(url), absl::nullopt);
+  } else if (provider == mojom::OnRampProvider::kRamp) {
+    url = base::StringPrintf(kRampBuyUrl, address.c_str(), symbol.c_str(),
+                             amount.c_str(), currency_code.c_str(), kRampID);
+    std::move(callback).Run(std::move(url), absl::nullopt);
+  } else if (provider == mojom::OnRampProvider::kSardine) {
+    auto internal_callback =
+        base::BindOnce(&AssetRatioService::OnGetSardineAuthToken,
+                       weak_ptr_factory_.GetWeakPtr(), chain_id, address,
+                       symbol, amount,  // Passing chainId as network
+                       currency_code, std::move(callback));
+    GURL sardine_token_url = GURL(kSardineClientTokensURL);
+    const std::string sardine_client_id(SARDINE_CLIENT_ID);
+    const std::string sardine_client_secret(SARDINE_CLIENT_SECRET);
+    std::string payload = base::StringPrintf(
```

Comment:
json should be created using base::Value to ensure proper formatting, etc... similar to url above